Add Ubuntu steps to delete blockchain and restart with PoET consensus

chenette opened the pull request:
Changed Ubuntu multi-node (network) procedure to delete the existing genesis block and blockchain data so that PoET consensus is specified in the genesis block.Added the steps to set PoET-sim as the consensus. Signed-off-by: Anne Chenette <chenette@bitwise.io>

askmish commented on docs/source/app_developers_guide/creating_sawtooth_network.rst:
```diff
@@ -258,16 +259,68 @@ a socket" in the `zmq-tcp API Reference <http://api.zeromq.org/4-2:zmq-tcp>`_.
 
-   If the first validator node is not running, start all Sawtooth components
-   except for the validator (see :doc:`docker`).
+#. If necessary, regenerate a validator private/public key pair in
```
keypairs needn't be regenerated at all, if they have had an existing sawtooth network in dev mode or they've followed steps in app developer setup. This needs to be clarified.

askmish commented on docs/source/app_developers_guide/creating_sawtooth_network.rst:
```diff
@@ -258,16 +259,68 @@ a socket" in the `zmq-tcp API Reference <http://api.zeromq.org/4-2:zmq-tcp>`_.
+
+#. Delete any existing blockchain data by removing all files from
+   ``/var/lib/sawtooth``.
+
```
Optionally, the logs in /var/log/sawtooth could be removed as well.

askmish commented on docs/source/app_developers_guide/creating_sawtooth_network.rst:
```diff
@@ -289,8 +342,13 @@ Step 1: Configure the Network on the First Node
-#. If the XO transaction processor is not running, start it as described in
-   Step 7 of :doc:`docker`.
+#. If necessary, start the other Sawtooth components.
+
```
settings-tp must be started as well.

chenette replied:
Answering my own question: It's clearer to stop all Sawtooth components in step 1, then restart them all here. That allows me to add a note that settings-tp and poet-validator-registry-tp are required, and that intkey-tp-python and xo-tp-python are "optional".

chenette commented:
Made the requested changes, plus a number of rewrites for clarity and some formatting fixes.  Kept this as a separate commit so it would be easier to see what changed (and to preserve the commit comments).

askmish commented on docs/source/app_developers_guide/creating_sawtooth_network.rst:
```diff
@@ -258,16 +268,78 @@ a socket" in the `zmq-tcp API Reference <http://api.zeromq.org/4-2:zmq-tcp>`_.
+      $ sawset proposal create -k /etc/sawtooth/keys/validator.priv \
+      -o config.batch
+      sawtooth.consensus.algorithm=poet \
+      sawtooth.poet.report_public_key_pem={simulator_rk_pub.pem} \
```
This should be:  `sawtooth.poet.report_public_key_pem=$(cat /etc/sawtooth/simulator_rk_pub.pem)`

askmish commented on docs/source/app_developers_guide/creating_sawtooth_network.rst:
```diff
@@ -289,16 +361,48 @@ Step 1: Configure the Network on the First Node
+
+   .. code-block:: console
+
+      $ sudo -u sawtooth sawtooth-rest-api -v
```
This command will use default validator component url. i.e. 127.0.0.1:4004
Could we add a note that the user would have to specify the validator component url, if its not the default one.

Refer: line 353:        --bind component:(component-bind-string} \

chenette commented on docs/source/app_developers_guide/creating_sawtooth_network.rst:
```diff
@@ -21,10 +22,14 @@ The following figure shows an example network with two validator nodes:
-multiple-node network.
+processing and static peering. However, it has the following differences:
+
+* The consensus is PoET instead of dev mode. This consensus type is also called
```
Good info.  The docs sometimes use the convention "PoET-SGX" and "PoET-sim" (or "PoET simulator"), but not consistently.  In other places, it's not clear which is meant.  I shall clarify in this procedure, and put the others on my to-do list. Thanks!

chenette commented:
@sofiter - Comments on the documentation are very welcome.  However, this PR is focused on the changes for "Creating a Sawtooth Network".  The best place for your comment is one of the following:

- Hyperledger's #sawtooth channel on Rocket.Chat: 
  https://chat.hyperledger.org/channel/sawtooth
  My handle is @achenette

- Hyperledger Sawtooth mailing list - subscribe here: 
  http://lists.hyperledger.org/mailman/listinfo/hyperledger-stl

Please repeat your comment in either Rocket.Chat or on the mailing list.  I'll be sure to check for your comment, because I want to ask you for more information about the doc version and environment you're using.  And thanks for your help with improving the Sawtooth documentation!

chenette commented on docs/source/app_developers_guide/creating_sawtooth_network.rst:
```diff
@@ -258,16 +270,78 @@ a socket" in the `zmq-tcp API Reference <http://api.zeromq.org/4-2:zmq-tcp>`_.
+
+   .. code-block:: console
+
+      $ sawset genesis -k /etc/sawtooth/keys/validator.priv -o config-genesis.batch
```
This step is lifted from the single-node procedure, which has you run it as "you" (not sawtooth or root).  The later "sawadm genesis" is run as root, but not this one.  See "Step 3: Create the Genesis Block":
"https://sawtooth.hyperledger.org/docs/core/nightly/master/app_developers_guide/ubuntu.html

If the "sawset genesis" step is wrong, it's wrong in both the single-node and multi-node procedures.  Can anyone confirm?

chenette commented on docs/source/app_developers_guide/creating_sawtooth_network.rst:
```diff
@@ -289,16 +363,60 @@ Step 1: Configure the Network on the First Node
 
         --peers tcp://172.0.1.3,tcp://172.0.1.4
```
Agreed.  I assume that this is the "public endpoint string", aka the ``--endpoint`` option, for the OTHER validator node(s). So, for this current example, I think it would be something like this:

     --peers tcp://172.0.1.3:8800,tcp://172.0.1.4:8800

Question: Would it be more helpful to show pretend host names instead of example IP addresses?  Such as: 

--peers tcp://MyPeer2:8800,tcp://MyPeer3:8800

chenette commented:
Added a fixup commit to fix errors, make requested changes, and clarify the information - especially the bind, endpoint, and peers info in the sawtooth-validator command.  

- Clarified the bind, endpoint, and peers options in Prequisites 

- Added a missing backslash in multi-line "sawset proposal create" command

- Clarified the example of the sawtooth-validator command (text and option order)                         
                                                                         
- Added the missing port numbers for the --peers example that's shown after the 
  sawtooth-validator command 

- Fixed a curly brace in the "boilerplate" sawtooth-validator command
    
- Corrected the option to specify a URL when starting the REST API and TPs
  (it's --connect, not --url)

- Moved the "start REST API" command to its own step

- Fixed wording, note location, and doc links throughout the Ubuntu procedure
